api/src/pcapi/routes/backoffice/offers/blueprint.py

```diff
@@ -724,17 +730,127 @@ def get_offer_details(offer_id: int) -> utils.BackofficeResponse:
+    return bool(
+        db.session.query(
+            offers_models.Stock.id,
+        )
+        .join(
+            bookings_models.Booking,
+            offers_models.Stock.bookings,
+        )
+        .join(offers_models.Offer, offers_models.Stock.offer)
+        .filter(
+            offers_models.Offer.isEvent,
+            offers_models.Stock.id == stock_id,
+            bookings_models.Booking.status.in_(
+                (
+                    bookings_models.BookingStatus.CONFIRMED,
+                    bookings_models.BookingStatus.USED,
+                )
+            ),
+        )
+        .count()
+    )
+
+
+@list_offers_blueprint.route("/<int:offer_id>/stock/<int:stock_id>/edit", methods=["POST"])
+@utils.permission_required(perm_models.Permissions.MANAGE_OFFERS)
+def offer_stock_edit(offer_id: int, stock_id: int) -> utils.BackofficeResponse:
+    if finance_api.are_cashflows_being_generated():
+        flash("Le script de génération des cashflows est en cours, veuillez réessayer plus tard.", "warning")
+        return redirect(request.referrer or url_for("backoffice_web.offer.get_offer_details", offer_id=offer_id), 303)
+    if not _is_stock_editable(stock_id):
+        flash("Ce stock n'est pas éditable.", "warning")
+        return redirect(request.referrer or url_for("backoffice_web.offer.get_offer_details", offer_id=offer_id), 303)
+
+    form = forms.EditStockForm()
+    stock = offers_models.Stock.query.filter_by(id=stock_id).one()
+    old_price = stock.price
+
+    if not form.validate():
+        flash("Le formulaire n'est pas valide", "warning")
+        return redirect(request.referrer or url_for("backoffice_web.offer.get_offer_details", offer_id=offer_id), 303)
+
+    if stock.price < form.price.data:
+        flash("Le prix ne doit pas être supérieur au prix original du stock.", "warning")
+        return redirect(request.referrer or url_for("backoffice_web.offer.get_offer_details", offer_id=offer_id), 303)
+
+    offers_api.update_used_stock_price(stock=stock, new_price=form.price.data)
+
+    logger.info(
+        "Un administrateur a changé le prix d'un stock d'évènement passé",
+        extra={
+            "user_id": current_user.id,
+            "stock_id": stock_id,
+            "old_price": old_price,
+            "new_price": form.price.data,
+        },
+    )
+    flash(f"Le stock {stock_id} a bien été mis à jour.", "success")
+    db.session.commit()
+    return redirect(url_for("backoffice_web.offer.get_offer_details", offer_id=offer_id), 303)
+
+
+@list_offers_blueprint.route("/<int:offer_id>/stock/<int:stock_id>/edit", methods=["GET"])
+@utils.permission_required(perm_models.Permissions.MANAGE_OFFERS)
+def get_offer_stock_edit_form(offer_id: int, stock_id: int) -> utils.BackofficeResponse:
+    if finance_api.are_cashflows_being_generated():
+        flash("Le script de génération des cashflows est en cours, veuillez réessayer plus tard.", "warning")
+        raise NotFound()
```

Comment:
L'erreur 404 ne me semble pas adaptée à la situation, d'autant plus que c'est surprenant de la coupler à un message flash. Selon moi, le plus adapté dans cette situation, puisqu'on a déjà ouvert une modale, serait d'y mettre le contenu : 
```suggestion
        return render_template(
            "components/turbo/modal_form.html",
            div_id=f"edit-offer-modal-{stock_id}",
            title=f"Baisser le prix du stock {stock_id}",
            alert="Le script de génération des cashflows est en cours, veuillez réessayer plus tard."
        )
```
Dans ce cas on devrait avoir l'erreur, pas de formulaire, et seul le bouton d'annulation.